File: app/src/main/java/com/cyder/atsushi/youtubesync/json_data/CreateRoom.java
package com.cyder.atsushi.youtubesync.json_data;

/**
 * Created by atsushi on 2017/10/27.
 */

public class CreateRoom extends JsonParameter{
    public Room room;

    public CreateRoom(String name, String description) {
        room = new Room(name, description);
    }

    private class Room {
        public String name;
        public String description;

        public Room(String name, String description) {
            this.name = name;
            this.description = description;
        }
    }
}

File: app/src/main/kotlin/com/cyder/atsushi/youtubesync/api/request/CreateRoom.kt
package com.cyder.atsushi.youtubesync.api.request

/**
 * Created by chigichan24 on 2018/03/31.
 */

class CreateRoom(name: String, description: String) {
    private val room = Room(name, description)
}

private data class Room (
        val name: String,
        val description: String
)